src/sentry/integrations/slack/requests/base.py

```diff
@@ -197,8 +210,40 @@ def authorize(self) -> None:
         verification_token = options.get("slack.verification-token")
 
         if signing_secret:
-            if self._check_signing_secret(signing_secret):
-                return
+            if self._organization and features.has(
```

Comment:
any chance we can move stuff into smaller functions?

Reply:
i refactored it a bit